packages/editor/src/components/widgets/code-editor/ScriptArea.tsx

```diff
@@ -15,9 +17,22 @@ type ScriptAreaProps = CodeEditorAreaProps & {
 
 const ScriptArea = (props: ScriptAreaProps) => {
   const browser = useBrowser();
-  const { setEditor, modifyEditor, getMonacoSelection } = useMonacoEditor({ scriptAreaDatatypeEditor: true });
+  const { setEditor, modifyEditor, getMonacoSelection, getCursorPosition, getSelectionRange } = useMonacoEditor();
+  const { isFocusWithin, focusWithinProps, focusValue } = useOnFocus(props.value, props.onChange);
   const path = usePath();
 
+  useEffect(() => {
+    const handleKeyDown = (event: KeyboardEvent) => {
+      if (event.key === 'F2' && isFocusWithin) {
+        event.preventDefault();
+        props.maximizeState.setIsMaximizedCodeEditorOpen(true);
+      }
+    };
+    document.addEventListener('keydown', handleKeyDown);
+    return () => {
+      document.removeEventListener('keydown', handleKeyDown);
+    };
+  }, [isFocusWithin, props.maximizeState]);
```

Comment:
did you checkout the other code editors? they already have a concept for keybindings... I would solve it more like those